plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -92,18 +97,26 @@ class FeatureDevSessionContext(val project: Project) {
 
     suspend fun zipFiles(projectRoot: VirtualFile): File = withContext(getCoroutineBgContext()) {
         val files = mutableListOf<VirtualFile>()
+        var totalSize: Long = 0
+
         VfsUtil.visitChildrenRecursively(
             projectRoot,
             object : VirtualFileVisitor<Unit>() {
-                override fun visitFile(file: VirtualFile): Boolean {
-                    if (file.isFile) {
-                        files.add(file)
-                        return true
-                    }
-                    return runBlocking {
-                        !ignoreFile(file, this)
+                override fun visitFile(file: VirtualFile): Boolean =
+                    runBlocking {
```

Comment:
Do we need to block the thread for the entire function? We should try to avoid using runBlocking in non-test functions. Im guessing we need it here for ignoreFile(). maybe we could change it to
val isFileIgnored = runBlocking {return ignoreFile....}
if(....)

Reply:
Yeah good tip, i did it in the next rev